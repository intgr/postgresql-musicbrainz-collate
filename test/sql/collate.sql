CREATE EXTENSION musicbrainz_collate;

-- Warning: Older ICU versions (before 4.8 or 4.6) may give different results
SELECT * FROM unnest('{AAA,äää,ÄÄÄ,aaa}'::text[]) AS txt ORDER BY musicbrainz_collate(txt);
SELECT musicbrainz_collate('foobar');
SELECT musicbrainz_collate('ľščťžýáí');
SELECT musicbrainz_collate('');
SELECT musicbrainz_collate(null);

-- Additional tests from Unicode Technical Report #10
-- http://www.unicode.org/reports/tr10/
SELECT * FROM unnest(E'{
    de luge,
    de Luge,
    de\u002dluge,
    de\u002dLuge,
    de\u2010luge,
    de\u2010Luge,
    death,
    deluge,
    deLuge,
    demark,
    ☠happy,
    ☠sad,
    ♡happy,
    ♡sad
}'::text[]) AS txt ORDER BY musicbrainz_collate(txt);

SELECT * FROM unnest('{
    role,
    roles,
    rule,
    rôle,
    Role,
    rôle,
    “role”,
    ro□le
}'::text[]) AS txt ORDER BY musicbrainz_collate(txt);
