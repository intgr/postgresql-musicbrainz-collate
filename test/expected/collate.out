CREATE EXTENSION musicbrainz_collate;
-- Warning: Older ICU versions (before 4.8 or 4.6) may give different results
SELECT * FROM unnest('{AAA,äää,ÄÄÄ,aaa}'::text[]) AS txt ORDER BY musicbrainz_collate(txt);
 txt 
-----
 AAA
 ÄÄÄ
 aaa
 äää
(4 rows)

SELECT musicbrainz_collate('foobar');
               musicbrainz_collate                
--------------------------------------------------
 \x283f0f283f18283f18283f0b283f0a283f1b010a010a00
(1 row)

SELECT musicbrainz_collate('ľščťžýáí');
                                    musicbrainz_collate                                     
--------------------------------------------------------------------------------------------
 \x283f15283f1c283f0c283f1d283f23283f22283f0a283f120186978697869786978697868d868d868d011400
(1 row)

SELECT musicbrainz_collate('');
 musicbrainz_collate 
---------------------
 \x010100
(1 row)

SELECT musicbrainz_collate(null);
 musicbrainz_collate 
---------------------
 
(1 row)

-- Additional tests from Unicode Technical Report #10:
-- http://www.unicode.org/reports/tr10/
SELECT * FROM unnest(E'{
de luge,
de Luge,
de\u002dluge,
de\u002dLuge,
de\u2010luge,
de\u2010Luge,
death,
deluge,
deLuge,
demark,
☠happy,
☠sad,
♡happy,
♡sad
}'::text[]) AS txt ORDER BY musicbrainz_collate(txt);
   txt   
---------
 ☠happy
 ☠sad
 ♡happy
 ♡sad
 de‐Luge
 de‐luge
 de Luge
 de luge
 de-Luge
 de-luge
 deLuge
 death
 deluge
 demark
(14 rows)

SELECT * FROM unnest('{
    role,
    roles,
    rule,
    rôle,
    Role,
    rôle,
    “role”,
    ro□le
}'::text[]) AS txt ORDER BY musicbrainz_collate(txt);
  txt   
--------
 “role”
 Role
 ro□le
 role
 rôle
 rôle
 roles
 rule
(8 rows)

